src/extensions/registries/workloads-overview-detail-registry.ts

```diff
@@ -19,22 +19,29 @@
  * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
 
+import { orderBy } from "lodash";
 import type React from "react";
 import { BaseRegistry } from "./base-registry";
 
 export interface WorkloadsOverviewDetailComponents {
-  Details: React.ComponentType<any>;
+  Details: React.ComponentType<{}>;
 }
 
 export interface WorkloadsOverviewDetailRegistration {
   components: WorkloadsOverviewDetailComponents;
   priority?: number;
 }
 
-export class WorkloadsOverviewDetailRegistry extends BaseRegistry<WorkloadsOverviewDetailRegistration> {
+type RegisteredWorkloadsOverviewDetail = Required<WorkloadsOverviewDetailRegistration>;
+
+export class WorkloadsOverviewDetailRegistry extends BaseRegistry<WorkloadsOverviewDetailRegistration, RegisteredWorkloadsOverviewDetail> {
   getItems() {
-    const items = super.getItems();
+    return orderBy(super.getItems(), "priority", "desc");
+  }
+
+  protected getRegisteredItem(item: WorkloadsOverviewDetailRegistration): RegisteredWorkloadsOverviewDetail {
+    const { priority = 50, ...rest } = item;
 
-    return items.sort((a, b) => (b.priority ?? 50) - (a.priority ?? 50));
+    return { priority, ...rest };
```

Comment:
this looks unrelated to the issue, did you test that it still works?

Reply:
Yes I did. With the kube-resource-map extension installed (and the default overview doesn't have `priority` defined) the order is the exact same as before.